### DISCUSSION TITLE
default the label name to the variable name

### DISCUSSION THREAD

**Discussion by aviau** [gleam-lang/gleam#3330] [Ideas & suggestions]
I am a fan of labels because I think they make the call-site way more readable.

However, I think they also make the function definitions very redundant. Its nice to be able to label arguments with a different name than the variable, but most if the time I prefer to just use the same name.

This results in many function definitions looking like this

```gleam
fn do_something(a a, b b) ...
```

Would it make sense to eithier:
1. Make it so that all parameters have labels, and default it to the variable name
    - The downside is that some function authors might not want to expose names and prefer the function to be called via posargs

2. Provide a syntax that will allow function authors to indicate they want to use variable names as label names
    - Something python-esque would be:
        * `fn do_something(*, a, b)` where all arguments after `*` are labeled.
        * `fn do_something(*, a, c b)` would still be valid, where `a` is labeled `a` and `b` is labeled `c`.

**Comment by lpil** [comment]
Hello! Please see previous discussion as to why this is not suitable for Gleam. In short, APIs must always be intentionally designed in Gleam, and we don't value brevity.

**Comment by sporto** [comment]
After writting a decent amount of gleam, I'm finding myself with a lot of duplication for adding labeled arguments. Labeled arguments are a big win for making the code more readable and avoiding bugs where you pass the arguments in the wrong order.

So, automatic labeled arguments would be great.
